src/node/internal/internal_fs.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  UV_DIRENT_DIR,
+  UV_DIRENT_FILE,
+  UV_DIRENT_BLOCK,
+  UV_DIRENT_CHAR,
+  UV_DIRENT_LINK,
+  UV_DIRENT_FIFO,
+  UV_DIRENT_SOCKET,
+} from 'node-internal:internal_fs_constants';
+import { getOptions } from 'node-internal:internal_fs_utils';
+import { validateFunction, validateUint32 } from 'node-internal:validators';
+import { ERR_MISSING_ARGS } from 'node-internal:internal_errors';
+
+const kType = Symbol('type');
+
+export class Dirent {
+  public name: string;
+  public parentPath: string;
+  public [kType]: number;
```

Comment:
Is this exposed in nodejs implementation as well?

Reply:
It's visible when the Dirent is inspected but otherwise no because the symbol is not exported. I made it private.